add-on/chrome/bootstrap.js

```diff
@@ -641,10 +647,43 @@ var WindowListener = {
               this._maybeShowBrowserSharingInfoBar();
             }
             break;
+          case "mousemove":
+            this.handleMousemove(event);
+            break;
           }
       },
 
       /**
+       * Handles mousemove events from gBrowser.
+       */
+      handleMousemove: function(event) {
+        // Only update every so often.
+        let now = Date.now();
+        if (now - this.lastCursorTime < MIN_CURSOR_INTERVAL) {
+          return;
+        }
+        this.lastCursorTime = now;
+
+        // Skip the update if cursor is out of bounds or didn't move much.
+        let browserBox = gBrowser.selectedBrowser.boxObject;
+        let deltaX = event.screenX - browserBox.screenX;
+        let deltaY = event.screenY - browserBox.screenY;
+        if (deltaX < 0 || deltaX > browserBox.width ||
+            deltaY < 0 || deltaY > browserBox.height ||
+            (Math.abs(deltaX - this.lastCursorX) < MIN_CURSOR_DELTA &&
+             Math.abs(deltaY - this.lastCursorY) < MIN_CURSOR_DELTA)) {
+          return;
+        }
+        this.lastCursorX = deltaX;
+        this.lastCursorY = deltaY;
+
+        this.LoopAPI.broadcastPushMessage("CursorPositionChange", {
+          cursorX: (deltaX * 100) / browserBox.width,
+          cursorY: (deltaY * 100) / browserBox.height
```

Comment:
As noted for the issue of detecting whether the link generator has resized the window, instead of calculating a percent, we can just send (deltaX, deltaY, width, height) so the link clicker getting a cursor event immediately knows that the dimensions changed (and potentially has different letterboxing).

If we don't need to send the width or height; and we're sending a percent instead, let's just keep the number between 0 and 1 and call it something like `xRatio` or `ratioX`. `cursorX` will probably be easily confused for the absolute coordinates.